src/problem.hpp

```diff
@@ -835,56 +834,36 @@ namespace NodeGLPK {
             }
 
             virtual void ExecuteWithEmitter(const ExecutionProgressSender* sender, eventemitter_fn_r fn) override {
-                hookinfo_ = {nullptr, sender, fn};
-                TermHookGuard hookguard{&hookinfo_};
+                HookInfo info = {nullptr, sender, fn};
+                TermHookGuard hookguard{&info};
                 try {
-                    if (state) {
+                    glp_intopt_start(lp->handle, &ctx);
```

Comment:
Double check my work on this that it behaves exactly as it did before, but all in the worker thread instead of the pseudo trampoline between the worker and main thread (I'm a little concerned about thread-safety with the prior implementation)